api.go

```diff
@@ -9,22 +9,16 @@ import (
 	"strings"
 )
 
-// Session represents an open API session.
-type Session struct {
-	apikey  string
-	baseurl url.URL
+// apiRequester is the type of anything that can handle an API request.
+type apiRequester interface {
+	// request sends out a full API request.
+	request(endpoint string, mixins []string, params map[string][]string) (*json.RawMessage, error)
 }
 
-// NewSession constructs a new Session with the given API key.
-func NewSession(apikey string) (*Session, error) {
-	url, err := url.Parse(`https://ury.york.ac.uk/api/v2`)
-	if err != nil {
-		return nil, err
-	}
-	return &Session{
-		apikey:  apikey,
-		baseurl: *url,
-	}, nil
+// AuthedRequester answers API requests by making an authed API call.
+type authedRequester struct {
```

Comment:
Is it worth bothering to call it an "authedRequester" ? Why not just "Requester"?